### PR TITLE
Union of methods to generate CNF and  histogram in only one method

### PR DESCRIPTION
**Pull request by leonelmaia**

### CODE REVIEW

**Comment by jmferreiratech** [comment]
Please, use the commit template.

File: src/Packet.java [hunk 1]
```diff
@@ -11,7 +11,8 @@
 	private int id;
```
Comment (jmarcelof) [inline src/Packet.java]: This file doesn't need to be changed on this patch.

File: src/Evaluation.java [hunk 2]
```diff
@@ -24,73 +23,46 @@ public Evaluation(String inPath, String outPath, String rede,
...
-	 * Gera o arquivo para a confecção do Distribuição Espacial do Tráfego
-	 * Aceito
-	 */
-	public void makeHistAccepTraff() {
-		int nPcks[] = new int[nDot]; // contém as quantidades pacotes
-		double accepTraffs[] = new double[nDot];
-		double step = (this.getAccepTraffMax()+this.getAccepTraffMin()) / (double) nDot;
-		for (int i = 1; i < nDot; i++)
-			accepTraffs[i] = (double)(i+1)*step;
+	public void makeHist(double vectors[], String fileName) {
```
Comment (jmarcelof) [inline src/Evaluation.java]: Make private and let makeHistLat()/makeHistAccepTraff() use it.

File: src/Evaluation.java [hunk 3]
```diff
@@ -24,73 +23,46 @@ public Evaluation(String inPath, String outPath, String rede,
 		accTraffic = read.accTrafficStats();
 	}
 
-	/* Gera o arquivo para a confecção do Distribuição Espacial da Latência Pura */
-	public void makeHistLat() {
```
Comment (jmarcelof) [inline src/Evaluation.java]: Rename it to writeHistogramOfLatency

File: src/Evaluation.java [hunk 4]
```diff
@@ -24,73 +23,46 @@ public Evaluation(String inPath, String outPath, String rede,
...
-			else if(++i < nDot) nPcks[i]++;
-
-		HandleFiles.WriteFile(outPath + "//", "ED_Lat" + strOL, nPcks, lats, nDot);
-	}
-
-	/*
-	 * Gera o arquivo para a confecção do Distribuição Espacial do Tráfego
-	 * Aceito
-	 */
-	public void makeHistAccepTraff() {
```
Comment (jmarcelof) [inline src/Evaluation.java]: Rename it to writeHistogramOfAcceptedTraffic

File: src/Evaluation.java [hunk 5]
```diff
@@ -1,19 +1,18 @@
 import java.util.ArrayList;
```
Comment (jmarcelof) [inline src/Evaluation.java]: Squash these imports.

File: src/Evaluation.java [hunk 6]
```diff
@@ -24,73 +23,46 @@ public Evaluation(String inPath, String outPath, String rede,
...
 					+ " : " + getAccepTraffMean() + "/"
 					+ getAccepTraffStdDev() + " : " + getAccepTraffMax()
 					+ "]";
-
 			HandleFiles.WriteFile(outPath + "//", "Relatorio" + strOL, Relat);
 	}
 
-	/* geta a carga oferecida do subteste */
-	public int OfferedLoad() {
+	public int OfferedLoad()
```
Comment (jmarcelof) [inline src/Evaluation.java]: Put the open curly brace on the same line as the function declaration.

File: src/TrafficMeasurer.java [hunk 7]
```diff
@@ -2,22 +2,20 @@
 import java.io.*;
```
Comment (jmarcelof) [inline src/TrafficMeasurer.java]: After making the requested changes, you won't need this file's modifications.

**Comment by jmferreiratech** [comment]
Rebase your patch on top of upstream.

**Comment by jmferreiratech** [comment]
Review your patch. writeHistogramOfLatency()/writeHistogramOfAccepTraff()/makeHist() don't seem to be ok.

**Comment by jmferreiratech** [comment]
Are you going to add a commit changing TrafficMeasurer.makeCNFLat()/TrafficMeasurer.makeCNFAccepTraff()?

File: src/Evaluation.java [hunk 8]
```diff
@@ -23,68 +25,65 @@ public Evaluation(String inPath, String outPath, String rede,
...
+		for (int i = 1; i < nDot; i++) {
+			vectors[i] = (double)(i+1)*step;
+		}
 		int i = 0;
-		for(Package pck: pcks)
-			if(pck.latency() <= lats[i]) nPcks[i]++;
-			else if(++i < nDot) nPcks[i]++;
-
-		HandleFiles.writeToFile(outPath + File.pathSeparator + "ED_Lat" + strOL, nPcks, lats);
+		switch (fileName) {
```
Comment (leonelmaia) [inline src/Evaluation.java]: I used the switch to select the Package variable that will be updated inside the for.However,is this a optimized solution?.

Comment (jmarcelof) [inline src/Evaluation.java]: No, the code inside each case it the same! 
You should not iterate over the 'pcks' variable but over the 'vector' (which will be the latency or accepted traffic).

File: src/Evaluation.java [hunk 9]
```diff
@@ -23,68 +25,65 @@ public Evaluation(String inPath, String outPath, String rede,
...
 	}
 
-	/* Gera o arquivo para a confecção do Distribuição Espacial da Latência Pura */
-	public void makeHistLat() {
-		int nPcks[] = new int[nDot]; // contém as quantidades pacotes
-		double lats[] = new double[nDot]; // contém as latências distintas
-		double step = (Collections.max(latencies, null) + Collections.min(latencies, null)) / (double) nDot;
-		for(int i = 0; i < nDot; i++)
-			lats[i] = (double)(i+1)*step;
-		Collections.sort(pcks, new Package.ByLatencyComparator());
```
Comment (jmarcelof) [inline src/Evaluation.java]: You forgot to keep the sort with 'comparator'.

File: src/Evaluation.java [hunk 10]
```diff
@@ -1,13 +1,15 @@
+package src;
```
Comment (jmarcelof) [inline src/Evaluation.java]: Do not create a package.

File: src/Evaluation.java [hunk 11]
```diff
@@ -23,68 +23,53 @@ public Evaluation(String inPath, String outPath, String rede,
...
-		int i = 0;
-		for(Package pck: pcks)
-			if(pck.acceptedTraffic() <= accepTraffs[i]) nPcks[i]++;
-			else if(++i < nDot) nPcks[i]++;
-
-		HandleFiles.writeToFile(outPath + File.separator + "SD_AT" + strOL, nPcks, accepTraffs);
+	public void writeHistogramOfLatency() {
+		makeHist(latencies ,"ED_Lat", new Package.ByLatencyComparator());
+	}
+	public void writeHistogramOfAccepTraff() {
```
Comment (jmarcelof) [inline src/Evaluation.java]: Separate these functions by a newline.

File: src/Evaluation.java [hunk 12]
```diff
@@ -23,68 +23,53 @@ public Evaluation(String inPath, String outPath, String rede,
...
 	}
 
-	/* Gera o arquivo para a confecção do Distribuição Espacial da Latência Pura */
-	public void makeHistLat() {
-		int nPcks[] = new int[nDot]; // contém as quantidades pacotes
-		double lats[] = new double[nDot]; // contém as latências distintas
-		double step = (Collections.max(latencies, null) + Collections.min(latencies, null)) / (double) nDot;
-		for(int i = 0; i < nDot; i++)
-			lats[i] = (double)(i+1)*step;
+	public void makeHist(List<Double> vector ,String fileName, Comparator<Package> comparator){
```
Comment (jmarcelof) [inline src/Evaluation.java]: You should sort with the comparator that this method receives as parameter.

File: src/Evaluation.java [hunk 13]
```diff
@@ -23,68 +23,53 @@ public Evaluation(String inPath, String outPath, String rede,
...
+		for (int i = 1; i < nDot; i++) {
+			vectors[i] = (double)(i+1)*step;
+		}
 		int i = 0;
-		for(Package pck: pcks)
-			if(pck.latency() <= lats[i]) nPcks[i]++;
-			else if(++i < nDot) nPcks[i]++;
-
-		HandleFiles.writeToFile(outPath + File.separator + "SD_Lat" + strOL, nPcks, lats);
+			for(Double vec : vector) {
```
Comment (jmarcelof) [inline src/Evaluation.java]: Fix the code indentation. And prefer using 4-space instead of tabs.

File: src/Evaluation.java [hunk 14]
```diff
@@ -125,5 +110,4 @@ private double accepTraffStdDev() // desvio padrão
 		}
 		return -1.0;
 	}
-
-}
+}
```
Comment (jmarcelof) [inline src/Evaluation.java]: Could you put a newline at the end of file?

File: src/Evaluation.java [hunk 15]
```diff
@@ -23,68 +23,54 @@ public Evaluation(String inPath, String outPath, String rede,
...
 
-	/* Gera o arquivo para a confecção do Distribuição Espacial da Latência Pura */
-	public void makeHistLat() {
-		int nPcks[] = new int[nDot]; // contém as quantidades pacotes
-		double lats[] = new double[nDot]; // contém as latências distintas
-		double step = (Collections.max(latencies, null) + Collections.min(latencies, null)) / (double) nDot;
-		for(int i = 0; i < nDot; i++)
-			lats[i] = (double)(i+1)*step;
-		Collections.sort(pcks, new Package.ByLatencyComparator());
+	public void makeHist(List<Double> vector ,String fileName, Comparator<Package> comparator){
```
Comment (jmarcelof) [inline src/Evaluation.java]: It can be private.

File: src/Evaluation.java [hunk 16]
```diff
@@ -23,68 +23,54 @@ public Evaluation(String inPath, String outPath, String rede,
...
+		}
+		for(Double vec : vector) {
+			if(vec <= vectors[i]){
+				nPcks[i]++;
+			}
+			else if(++i < nDot){
+				nPcks[i]++;
+			}
+		}
+		HandleFiles.writeToFile(outPath + File.pathSeparator + fileName + strOL, nPcks, vectors);
```
Comment (jmarcelof) [inline src/Evaluation.java]: Replace all 'File.pathSeparator' by 'File.separator'.

**Comment by leonelmaia** [comment]
The changes requested are done.

File: src/TrafficMeasurer.java [hunk 17]
```diff
@@ -55,24 +55,26 @@ public void makeRelats() {
...
-			accepTraffmean[i] = OL[i].averageAccepTraff();
+			averageAccepTraff[i] = OL[i].averageAccepTraff();
 		}
-		HandleFiles.writeToFile(outPath + rede + File.separator + "CNF_AT", offerload, accepTraffmean);
+		makeCNF(new double[OL.length],averageAccepTraff, "CNF_AT");
 	}
 
-}
\ No newline at end of file
+	private void makeCNF (double [] offerload, double[] vectormean, String filename) {
```
Comment (jmarcelof) [inline src/TrafficMeasurer.java]: Do not pass 'offerload' as parameter, instantiate it (using the size of vectormean) inside that method.

File: src/TrafficMeasurer.java [hunk 18]
```diff
@@ -30,55 +31,51 @@ public TrafficMeasurer(String nome, String inPath, String outPath) {
...
+
+	private void makeCNFLat()  {
+		double[] averageLatency = null;
 		for (int i = 0; i < OL.length; i++) {
-			offerload[i] = OL[i].OfferedLoad() / 100.0;
-			latmean[i] = OL[i].averageLatency();
+			averageLatency[i] = OL[i].averageLatency();
 		}
-		HandleFiles.writeToFile(outPath + rede + File.separator + "CNF_Lat", offerload, latmean);
+	makeCNF(averageLatency , "CNF_Lat");
```
Comment (jmarcelof) [inline src/TrafficMeasurer.java]: Fix the indentation here.

**Pull request merged**